File: Clients/client/src/app/_modules/utilities/models/pagination.ts
export class Pagination{
    page:number=1;
    pageSize:number=10;
    totalItems:number=null
    totalPages:number=null

    constructor(){}

    castJsonToClass(json:Pagination){
        if(json.pageSize){
          this.pageSize = json.pageSize;
        }
        else{
          this.pageSize = 10;
        }
        if(json.page){
          this.page = json.page;
        }
        if(json.totalPages){
          this.totalPages = json.totalPages;
        }
        if(json.totalItems){
          this.totalItems = json.totalItems;
        }
      }
}

File: Services/Contractors/Contractors.Appilcation/Features/Contractors/Pagination.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Contractors.Appilcation.Features.Contractors
{
    public class Pagination
    {
        public Pagination() { }

        public Pagination(int count, int pageNumber, int pageSize)
        {
            TotalPages = (int)Math.Ceiling(count / (double)pageSize);
            Page = Math.Min(pageNumber, TotalPages);
            TotalCount = count;
            PageSize = pageSize;
        }
        public int Page { get; set; }
        public int TotalPages { get; set; }
        public int TotalCount { get; set; }
        public int PageSize { get; set; }
    }
}
